ngraph/test/type_prop/top_k.cpp

```diff
@@ -75,14 +77,43 @@ TYPED_TEST_P(topk_type_prop, topk_v1_partial_ouptut)
     {
         auto k = make_shared<op::Constant>(element::i32, Shape{}, 3);
         auto topk = make_shared<TypeParam>(data, k, 1, "max", "value");
+        EXPECT_EQ(topk->get_output_shape(0), Shape({2, 3}));
         EXPECT_EQ(topk->get_output_partial_shape(0), PartialShape({2, 3}));
     }
 }
 
+TYPED_TEST_P(topk_type_prop, topk_rank_static_k_unknown)
+{
+    const auto data_shape = Shape{1, 10, 100};
+    const auto data = make_shared<op::Parameter>(element::f32, data_shape);
+    const int64_t axis = 1;
+    const Dimension fully_dynamic_dimension{-1};
+    const PartialShape fully_dynamic_axis_shape{1, fully_dynamic_dimension, 100};
+
+    {
+        const auto k = make_shared<op::Parameter>(element::i32, PartialShape({}));
+        const auto topk = make_shared<TypeParam>(data, k, axis, "max", "value");
+
+        EXPECT_EQ(topk->get_output_partial_shape(0), fully_dynamic_axis_shape);
+    }
+    {
+        const int64_t c = 5;
+        const auto cnst = make_shared<op::v0::Constant>(element::i64, Shape{}, &c);
```

Comment:
`cnst` logically represents/holds the value of `k` and should be named that way. Also, the construction of a constant by passing a pointer to a local is very unusual, can you move the value of 5 directly to the constructor?